python/equistore-core/equistore/core/labels.py

```diff
@@ -492,6 +492,40 @@ def insert(self, index: int, name: str, values: np.ndarray) -> "Labels":
 
         return Labels(names=new_names, values=new_values)
 
+    def permute(self, dims: List[int]) -> "Labels":
+        """Permute dimensions according to ``dims`` in the :py:class:`Labels`.
+
+        :param dims: desired ordering of the dimensions
+        :raises ValueError: if length of ``dims`` does not match the Labels length
+        :raises ValueError: if duplicate values are present in ``dims``
+
+        >>> import numpy as np
+        >>> from equistore import Labels
+        >>> label = Labels(["foo", "bar", "baz"], np.array([[42, 10, 3]]))
+        >>> label
+        Labels(
+            foo  bar  baz
+            42   10    3
+        )
+        >>> label.permute([2, 0, 1])
+        Labels(
+            baz  foo  bar
+             3   42   10
+        )
+        """
+        if len(dims) != len(self.names):
+            raise ValueError(
+                f"number of dimensions Labels ({len(self.names)}) does not match the "
+                f"length of the desired ordering of dimensions ({len(dims)})."
+            )
+
+        names = [self.names[d] for d in dims]
+
+        if len(set(names)) != len(names):
+            raise ValueError("duplicate dims are not allowed")
```

Comment:
```suggestion
            raise ValueError("duplicate dimensions are not allowed")
```

Reply:
Also, do we need to check this anyway? Since the rust code will also do this check

Reply:
Ohh yes you are right!